.gitignore

```diff
@@ -1,4 +1,42 @@
+### Bazel ###
+# gitignore template for Bazel build system
+# website: https://bazel.build/
+
+# Ignore all bazel-* symlinks. There is no full list since this can change
+# based on the name of the directory bazel is cloned into.
 /bazel-*
 
-# Rust artifacts.
-/**/target
+### Rust ###
+# Generated by Cargo
+# will have compiled files and executables
+/**/target/
+
+# Remove Cargo.lock from gitignore if creating an executable, leave it for libraries
+# More information here https://doc.rust-lang.org/cargo/guide/cargo-toml-vs-cargo-lock.html
+Cargo.lock
+
+# These are backup files generated by rustfmt
+**/*.rs.bk
+
+### CLion ###
+# Covers JetBrains IDEs: IntelliJ, RubyMine, PhpStorm, AppCode, PyCharm, CLion, Android Studio and WebStorm
+# Reference: https://intellij-support.jetbrains.com/hc/en-us/articles/206544839
+
+# User-specific stuff
+.idea/**/workspace.xml
```

Comment:
Can we just ignore `.idea` entirely? Or do you think there is a reasonable use case for checking in some of this stuff?

Reply:
I think once we can make the project load in CLion properly it'd be reasonable to submit the .idea files. Anyways CLion is a Google supported IDE for C++, so it'd make sense.

Reply:
In my experience it is rarely a good idea to check in specific IDE or editor configuration files, as developers have different needs / preferences. That said, I do not know anything about CLion so I am happy to follow your advice is you think that whatever is going to eventually be checked in is sufficiently non-controversial that everyone using CLion would be happy with.

Reply:
SGTM.